celery/utils/time.py

```diff
@@ -383,3 +385,20 @@ def utcoffset(time=_time, localtime=_time.localtime):
 def adjust_timestamp(ts, offset, here=utcoffset):
     """Adjust timestamp based on provided utcoffset."""
     return ts - (offset - here()) * 3600
+
+
+def get_exponential_backoff_interval(
+    factor,
+    retries,
+    maximum,
+    full_jitter=False
```

Comment:
When I was writing this function, it seemed kind of odd that it should handle both full jitter and custom deviation. After looking at `tenacity` source code and reading the AWS blog post, I thought that perhaps retaining the simple Full Jitter algorithm, was the best choice here.

For example if you set a jitter deviation of 1:
- `factor=2, retries=1` yields a countdown of 4 , resulting in a delay in the range `[3, 5]`
- `factor=2, retries=4` yields a countdown of 32 , resulting in a delay in the range `[31, 33]`

So in the fourth retry, the jitter deviation is proportionately negligible and you may be actually increasing the possibility of e.g. HTTP calls to a service happening almost at the same time, whereas using a uniform distribution of the delay time of 32, and have let's say 10 concurrent tasks, there is a greater chance that they will be retried at a more timely manner.

```
# Full jitter
>>> sorted([random.randrange(32) for _ in range(10)])
>>> [7, 15, 16, 20, 22, 23, 25, 27, 29, 31]
# Custom deviation
>>> sorted([32 + random.randrange(2) for _ in range(10)])
>>> [32, 32, 32, 32, 32, 32, 33, 33, 33, 33]
```

I thought that it would be better to use the simple version for now, and get some user feedback. If you have an actual use case of a service that implements a different algorithm, please let me know.